addon/list-utils.js

```diff
@@ -0,0 +1,69 @@
+import * as utils from './utils'
+import Ember from 'ember'
+
+function promiseValue (value) {
+  return new Ember.RSVP.Promise(function (resolve) {
+    resolve(value)
+  })
+}
+
+/**
+ * setOptions - set a list's available options
+ * @param  {Object} modelDef the bunsen model definition
+ * @param  {Object} dbStore the ember-data store
+ * @returns {Promise} a promise that resolves to a list of items
+ */
+export function getOptions (modelDef, dbStore) {
+  const enumDef = modelDef.enum
+  const queryDef = modelDef.query
+  let result
+  if (enumDef) {
+    result = promiseValue(getEnumValues(enumDef))
+  } else if (queryDef) {
+    result = getAsyncDataValues(modelDef, queryDef, dbStore)
+  } else {
+    result = promiseValue([])
+  }
+  return result
```

Comment:
I would probably write it @sandersky's way too, but there are advantages to the original way, particularly if you want to debug what `result` is before it's returned.